services/fins/src/fins/core/information_service.py

```diff
@@ -0,0 +1,88 @@
+# limitations under the License.
+"""Contains logic for public file information storage, retrieval and deletion."""
+
+import logging
+
+import ghga_event_schemas.pydantic_ as event_schemas
+from hexkit.protocols.dao import ResourceNotFoundError
+
+from fins.adapters.inbound.dao import FileInformationDaoPort
+from fins.core.models import FileInformation
+from fins.ports.inbound.information_service import InformationServicePort
+
+log = logging.getLogger(__name__)
+
+
+class InformationService(InformationServicePort):
+    """A service that handles storage and deletion of relevant metadata for files
+    registered with the Internal File Registry service.
+    """
+
+    def __init__(self, *, file_infomation_dao: FileInformationDaoPort):
+        self._file_information_dao = file_infomation_dao
+
+    async def deletion_requested(self, file_id: str):
+        """Handle deletion requestes for information associated with the give file ID."""
+        try:
+            await self._file_information_dao.get_by_id(id_=file_id)
+        except ResourceNotFoundError:
+            log.info(
+                f"Information for file with id '{
+                    file_id}' has already been deleted."
+            )
+            return
+
+        await self._file_information_dao.delete(id_=file_id)
+        log.info(f"Successfully deleted entries for file with id '{file_id}'.")
+
+    async def register_information(
+        self, file_registered: event_schemas.FileInternallyRegistered
+    ):
+        """Store information for a file newly registered with the Internal File Registry."""
+        file_information = FileInformation(
+            file_id=file_registered.file_id,
+            size=file_registered.decrypted_size,
+            sha256_hash=file_registered.decrypted_sha256,
+        )
+        file_id = file_information.file_id
+
+        try:
+            existing_information = await self._file_information_dao.get_by_id(
+                id_=file_id
+            )
+            log.debug(f"Found existing information for file {file_id}")
+            # Only log if information to be inserted is a mismatch
+            if existing_information != file_information:
+                information_exists = self.MismatchingInformationAlreadyRegistered(
+                    file_id=file_id
+                )
+                log.error(information_exists)
+        except ResourceNotFoundError:
+            await self._file_information_dao.insert(file_information)
+            log.debug(f"Sucessfully inserted information for file {file_id} ")
+
+    async def serve_information(self, file_id: str) -> FileInformation:
+        """Retrieve stored public information for the five file ID to be served by API."""
+        try:
+            file_information = await self._file_information_dao.get_by_id(file_id)
+            log.debug(f"Information for file {
+                      file_information.file_id} has been served.")
```

Comment:
```suggestion
                      file_information.file_id} has been found.")
```
Maybe tweak this for clarity. Could also add ` -- returning info to API` or something similar if you think it would be useful

Reply:
Just a debug message, so I'd leave out the API part.